### PR TITLE
Control form error handling eb

### LINKED ISSUE #39: Error handling: user adds incomplete guest details (missing name/enough #s for phone #)

### PR DESCRIPTION
**Pull request by ericberglund117**
# Pull Request

## What's this PR do? 

- [x] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Testing
- [x] Styling
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Description of changes?
- User can input guest name and phone number 
- creates automatic dashes in phone number input 
- Addresses error handling and edge cases like if a user:
- only inputs a name 
- only inputs a phone number
- doesn't input a complete phone number 
- or user tries to input a phone number that doesn't contain a numerical value. 
- create a 'clear guest and phone number inputs' button 

## Any background context you want to provide? 
- There is a bug in deleting the phone number via backspace. It always leaves the first character 
- After hours of debugging, there was no solution so I had to create clear button in order to solve this problem 

## What are the relevant tickets? 
* Addresses
* Closes #39

### CODE REVIEW

File: src/components/GuestList/GuestList.tsx [hunk 1]
```diff
@@ -8,14 +8,40 @@ import { StyledButton, StyledCard } from '../App/styledComponents.styles'
...
+      setErrorMessage('')
+    } else {
+      setIsError(true)
+      setErrorMessage('Phone number only accepts numerical values')
+    }
+  } 
+
+  const formatPhoneText = (value: string) => {
+    if (value.length > 3 && value.length <= 6) {
+      value = value.slice(0,3) + "-" + value.slice(3);
```
Comment (hannakim91) [inline src/components/GuestList/GuestList.tsx:37]: This helps a lot with user experience - not seeing one long string of numbers

File: src/components/GuestList/GuestList.tsx [hunk 2]
```diff
@@ -54,31 +93,32 @@ const GuestList: React.FC = () => {
...
         />
-        <StyledButton onClick={event => submitGuest(event)}>
+        <StyledButton>
+          <div id="translate"></div>
+          <a className="link" id="addListButton" onClick={event => submitGuest(event)}>Add To Guest List</a>
+        </StyledButton>
+        <StyledButton>
           <div id="translate"></div>
-            <a className="link" id="addListButton">Add To Guest List</a>
+          <a className="link" id="clearFormButton" onClick={event => clearInputs()}>Clear Guest and Phone Number</a>
```
Comment (hannakim91) [inline src/components/GuestList/GuestList.tsx:109]: This seems like a good workaround for that strange bug we had not being able to clear the form input entirely!

File: src/weddingData.tsx [hunk 3]
```diff
@@ -44,22 +44,20 @@ export const individualWedding = {
...
-  photoList: []
-  // [
-  //   {photoId: 1, guests: ["Bob", "Sue", "Mary"], description: "college buddies"},
-  //   {photoId: 2, guests: ["Gary", "Jim", "Mary", "Bob"], description: "The Out-laws"},
-  //   {photoId: 3, guests: ["John", "Bob", "Mary"], description: "The In-laws"},
-  //   {photoId: 4, guests: ["John", "Eric", "Brigette", "Bob"], description: "trouble"},
-  //   {photoId: 5, guests: ["Sue", "Gary", "John", "Jim", "Mary"], description: "old babysitters"},
-  //   {photoId: 6, guests: ["Bob", "Mary", "Sue", "John", "Jim", "Gary"], description: "this one's for mom"}
-  // ]
+  familyPhotoList: [
```
Comment (hannakim91) [inline src/weddingData.tsx:47]: We won't be needing our fake data soon!

**Review by hannakim91 — APPROVED** [review]

File: src/components/GuestList/GuestList.tsx [hunk 4]
```diff
@@ -8,14 +8,40 @@ import { StyledButton, StyledCard } from '../App/styledComponents.styles'
...
+    }
+  } 
+
+  const formatPhoneText = (value: string) => {
+    if (value.length > 3 && value.length <= 6) {
+      value = value.slice(0,3) + "-" + value.slice(3);
+    } else if(value.length > 6) {
+      value = value.slice(0,3) + "-" + value.slice(3,6) + "-" + value.slice(6);
+    }
+    setPhoneNumber(value)
```
Comment (Kristiannmiller) [inline src/components/GuestList/GuestList.tsx:41]: We might need to add a "+1" to the front of all the phone numbers for Twilio, but we will see when we get to that point

File: src/components/GuestList/GuestList.tsx (hunk 2 above)
Comment (Kristiannmiller) [inline src/components/GuestList/GuestList.tsx:109]: This works great for now! We'll keep looking for a solution, but this is a nice little bandaid for the time being

**Review by Kristiannmiller — APPROVED** [review]

**Pull request merged**